### ISSUE TITLE
[Bug report] please put your title here

### ISSUE THREAD

**Issue by icygreen-xy** [numbbo/coco#2189]
**Describe the bug**
I want to use a new virtual environment( new_env ) to run the coco benchmark, but I met a problem with " no module named cocoex".
Because my other package can not run well in the original environment( base ), so I use this new virtual environment to replace, but here I can not run coco benchmark well.

In my original environment (base), I can run it well in line1 and it does not show anything about " no module named cocoex".

How can I address this problem? Thanks a lot! 

**Your System (please complete the following information when it seems relevant):**
 - Operating system: macOS 13.2.1
 - COCO version: coco/code-experiments/build/python)/example_experiment2.py /
 - (I am not sure version )
 - Browser (if relevant):  safari 

base environment in conda -JupyterNotebook
![image](https://user-images.githubusercontent.com/110183414/227220043-5e1fffc0-695c-470a-858c-2a36c51b42ce.png)

new virtual environment in conda -JupyterNotebook
![image](https://user-images.githubusercontent.com/110183414/227220086-937a23d5-4365-410e-95db-1e62e712a120.png)

my conda environment
![image](https://user-images.githubusercontent.com/110183414/227220143-d98fddac-c9d4-4134-88ad-a8c771109e6b.png)

**Comment by nikohansen** [comment]
Did you run
```
python do.py run-python
```
in this environment in the `coco` root folder git checkout?

**Comment by icygreen-xy** [comment]
yes, I did
also I reinstall pip install cocopp -point 1 to this point 3 
but it showed:
![image](https://user-images.githubusercontent.com/110183414/227566774-0daf4d59-41e6-4145-baf4-53923eb55c23.png)

![image](https://user-images.githubusercontent.com/110183414/227566852-85ffe364-7bfd-492d-aef2-ce29118a33ff.png)

**Comment by nikohansen** [comment]
Right, so the installation didn't succeed and as explained in the message you should add "user-install" to the command.

**Comment by nikohansen** [comment]
Right, so the installation didn't succeed. I suspect this is a missing dependency problem (Cython?). Did you install all necessary dependencies? [Apparently, `python-dev` needs to be installed](https://github.com/numbbo/coco#programming-language-specifics--) to install `cocoex` but this also depends on the OS you are working with.

**Comment by icygreen-xy** [comment]
yeah, thanks a lot 

I've checked my new virtual environment, and there is an existing Cypthon.
But I can not find `python-dev` in this virtual environment.


**Cypthon:**
![image](https://user-images.githubusercontent.com/110183414/227697200-a58ca311-486f-467c-a7c8-af6a1343ca45.png)

**python-dev**
![image](https://user-images.githubusercontent.com/110183414/227700469-a6565194-e89a-4760-aeff-d408beb64c61.png)
![image](https://user-images.githubusercontent.com/110183414/227700477-99f5997a-d3b4-4b77-b6bc-9c6576adc2d5.png)
![image](https://user-images.githubusercontent.com/110183414/227700492-2e593db0-f38a-4527-aded-9a57cb0f40cf.png)

I am not so sure this one is `pythom-dev`?

![image](https://user-images.githubusercontent.com/110183414/227700703-b842194f-3f44-4fdb-b00a-3cc3d15fa0c8.png)

**Comment by nikohansen** [comment]
This looks more like a Python 3.11 problem than related to coco. See e.g.

https://stackoverflow.com/questions/74979674/gensim-install-in-python-3-11-fails-because-of-missing-longintrepr-h-file
https://github.com/aio-libs/aiohttp/issues/6600

Maybe try to update cython? The cocoex install works for me under macOS also with Python 3.11 (with a warning).

**Comment by icygreen-xy** [comment]
yeah
I've updated my cython now. But it can not work with 

> ModuleNotFoundError: No module named 'cocoex'

![image](https://user-images.githubusercontent.com/110183414/227710541-7100b7f1-e0ce-43ea-a5f1-0f008831cfed.png)

**Comment by nikohansen** [comment]
Just FTR, the original problem was the `#include` error
```
fatal error: 'longintrepr.h' file not found
```
We don't see whether it's still the same.

**Comment by icygreen-xy** [comment]
I am not so sure about how to deal with "Just FTR, the original problem was the #include error"

I've tried it in my base environment using the same file - `coco`, and it can run successfully in the base environment. 

When I create a new environment by `conda create --name new`, then it can not work with this bug.

**Comment by nikohansen** [comment]
1. We don't see what the latest error was (why the `do.py` built failed). It might be a different error than from including `longintrepr.h` now, but we don't know.
2. I suspect a simple fix could be to create a new Python 3.10 environment instead of a Python 3.11 environment.

**Comment by icygreen-xy** [comment]
I downgraded my version of Python 3.11 -> Python 3.10, and it made successfully work now in a virtual environment! 
Many thanks! :D
